modules/cluster-management/src/main/java/org/apache/ignite/internal/cluster/management/raft/CmgRaftService.java

```diff
@@ -228,6 +228,38 @@ public Set<String> nodeNames() {
                 .collect(toSet());
     }
 
+    /**
+     * Returns a set of consistent IDs of the majority of the voting nodes of the CMG, including a leader.
+     *
+     * @return Set of consistent IDs of the majority of the voting nodes of the CMG, including a leader.
+     */
+    public CompletableFuture<Set<String>> majority() {
+        Peer leader = raftService.leader();
+
+        if (leader == null) {
+            return raftService.refreshLeader().thenCompose(v -> majority());
+        }
+
+        List<Peer> peers = raftService.peers();
+
+        assert peers != null;
+
+        int peersCount = peers.size();
+        String leaderId = leader.consistentId();
+
+        // Take half of the voting peers without the leader.
+        Set<String> result = peers.stream()
+                .map(Peer::consistentId)
+                .filter(consistentId -> !consistentId.equals(leaderId))
+                .limit(peersCount / 2)
+                .collect(toSet());
```

Comment:
It is generally a bad idea to modify a set, returned by `toSet()`, there's no guarantee on its mutability

Reply:
Also, why is it a set and not a list?